### PR TITLE
Added error codes utility runner workflow

### LINKED ISSUE #213: [CI] add error codes utility runner workflow
#### Current Behavior

Error codes and their reference is not being updated.
#### Desired Behavior

Add the error codes updater workflow to update error codes and publish error codes reference

#### Implementation

1. Create the error codes updater workflow for the repo. Modify it a bit according to the repository you're working on.
Sample: https://github.com/layer5io/meshkit/blob/master/.github/workflows/error-codes-updater.yaml
2. Delete `errorutil_analyze_errors.json` `errorutil_analyze_summary.json` `errorutil_errors_export.json`
3. Add the three files from previous step to `.gitignore`
Sample: https://github.com/layer5io/meshkit/blob/5bc1f9c13e60c7674d8b8c1d8ea434cbb2ce08fd/.gitignore#L21-L23
4. Also remove them from cache using 
`git rm --cached errorutil_analyze_errors.json errorutil_analyze_summary.json errorutil_errors_export.json`
5. Move `component_info.json` to a `helpers` directory.
6. Update command for running utility in Makefile to :
```make
	error:
	go run github.com/layer5io/meshkit/cmd/errorutil -d . analyze -i ./helpers -o ./helpers
```

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

**Description**

Added error codes utility runner workflow

This PR fixes #213 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/error-ref-publisher.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Error Code Reference Publisher
```
Comment (DelusionalOptimist) [inline .github/workflows/error-ref-publisher.yaml:1]: @sayantan1413 I've made a few changes to the one currently in meshkit. Please take a look and update accordingly. Thanks : ). https://github.com/DelusionalOptimist/meshkit/blob/error-codes-final/.github/workflows/error-codes-updater.yaml

File: .gitignore [hunk 2]
```diff
@@ -18,3 +18,6 @@ log.txt
 
 .idea
 
+cmd/errorutil/errorutil
+/errorutil
```
Comment (DelusionalOptimist) [inline .gitignore:22]:
```suggestion
**errorutil
**errorutil_analyze_errors.json
**errorutil_analyze_summary.json
**errorutil_errors_export.json
```

File: linkerd/error.go [hunk 3]
```diff
@@ -7,27 +7,27 @@ import (
 
 var (
 	// ErrInstallLinkerdCode is the error code for ErrInstallLinkerd
-	ErrInstallLinkerdCode = "1003"
+	ErrInstallLinkerdCode = "replace"
```
Comment (DelusionalOptimist) [inline linkerd/error.go:10]: same here as well

File: internal/config/error.go [hunk 4]
```diff
@@ -19,9 +19,9 @@ import (
 )
 
 const (
-	ErrEmptyConfigCode           = "1000"
-	ErrGetLatestReleasesCode     = "1001"
-	ErrGetLatestReleaseNamesCode = "1002"
+	ErrEmptyConfigCode           = "replace"
+	ErrGetLatestReleasesCode     = "replace"
+	ErrGetLatestReleaseNamesCode = "replace"
```
Comment (DelusionalOptimist) [inline internal/config/error.go:24]: similarly here. you don't need to replace current error codes. its fine to keep them : )

File: helpers/component_info.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
```
Comment (DelusionalOptimist) [inline helpers/component_info.json:1]: no changes needed in `component_info.json`. Just move it to `helpers` directory unchanged.

**Review by DelusionalOptimist — CHANGES_REQUESTED** [review]
Thanks for this @sayantan1413.
Just some minor changes requested : )

File: helpers/component_info.json (hunk 5 above)
Comment (sayantan1413) [reply helpers/component_info.json:1]: Okay so like I should not make any changes in the error codes right?

Comment (DelusionalOptimist) [reply helpers/component_info.json:1]: yes

File: .github/workflows/error-ref-publisher.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      - "**.go"
+
+jobs:
+  error-ref:
+    name: Error codes utility
+    if: github.repository == 'meshery/meshery-linkerd'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        # token here with write access to linkerd repo
```
Comment (DelusionalOptimist) [inline .github/workflows/error-ref-publisher.yaml:16]:
```suggestion
        # token here with write access to meshery-linkerd repo
```

File: .github/workflows/error-ref-publisher.yaml [hunk 7]
```diff
@@ -0,0 +1,64 @@
+name: Meshkit Error Codes Utility Runner
+on:
+  push:
+    branches:
+      - "master"
```
Comment (DelusionalOptimist) [inline .github/workflows/error-ref-publisher.yaml:5]:
```suggestion
      - 'master'
```

File: .github/workflows/error-ref-publisher.yaml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    paths:
+      - "**.go"
```
Comment (DelusionalOptimist) [inline .github/workflows/error-ref-publisher.yaml:7]:
```suggestion
      - '**.go'
```

File: .github/workflows/error-ref-publisher.yaml [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Setup Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ secrets.GO_VERSION }}
+
+      - name: Run utility
+        run: |
+          go get github.com/layer5io/meshkit/cmd/errorutil
+          go run github.com/layer5io/meshkit/cmd/errorutil -d . update --skip-dirs meshery -i ./helpers -o ./helpers
+      # to update errorutil* files in meshkit repo
```
Comment (DelusionalOptimist) [inline .github/workflows/error-ref-publisher.yaml:30]:
```suggestion
      # to update errorutil* files in meshery-linkerd repo
```

**Review by DelusionalOptimist — CHANGES_REQUESTED** [review]

**Review by DelusionalOptimist — APPROVED** [review]
LGTM. Thanks @sayantan1413 :partying_face:

**Pull request merged**